sphinx_click/ext.py

```diff
@@ -162,14 +172,17 @@ def _format_options(ctx: click.Context) -> ty.Generator[str, None, None]:
         if isinstance(param, click.Option) and not getattr(param, 'hidden', False)
     ]
 
+    command_name = ctx.command_path.replace(' ', '-')
     for param in params:
-        for line in _format_option(param):
+        for line in _format_option(param, command_name):
```

Comment:
Instead of passing through this can you either (a) generate the label here, like we do for envvars in `_format_envvars` or (b) pass through the context (`ctx`) as the first parameter? We should also have a `_format_command_name(ctx)` helper to ensure we do the formatting consistently

Reply:
I do think that generating the label inside `_format_option()` makes more sense as it's part of the generated content for each specific option, which is what that function should be in charge of.
Passing the context is just as good though; I'll do that instead.
I agree about the extra function. I'll add that as well.

Reply:
> I do think that generating the label inside `_format_option()` makes more sense as it's part of the generated content for each specific option, which is what that function should be in charge of.

:+1: If we're doing this, can we do the same for `_format_envvar` (i.e. move the logic there from `_format_envvars`)

> Passing the context is just as good though; I'll do that instead. I agree about the extra function. I'll add that as well.

:+1: